core/class_db.cpp

```diff
@@ -1115,11 +1109,39 @@ MethodBind *ClassDB::bind_methodfi(uint32_t p_flags, MethodBind *p_bind, const c
 		ERR_EXPLAIN("Method already bound: " + instance_type + "::" + mdname);
 		ERR_FAIL_V(NULL);
 	}
+
 #ifdef DEBUG_METHODS_ENABLED
-	p_bind->set_argument_names(method_name.args);
-	p_bind->set_return_type(rettype);
+	Vector<StringName> arg_type_names = p_bind->get_argument_names();
+	Vector<StringName> arg_names = method_name.args;
+
+	if (arg_names.size() != arg_type_names.size()) {
+
+		if (arg_names.size() > arg_type_names.size()) {
+			memdelete(p_bind);
+			ERR_EXPLAIN("Method definition provides more arguments than the method actually has: " + instance_type + "::" + mdname);
+			ERR_FAIL_V(NULL);
+		}
+
+		// We need to have all the argument names in the case of MethodBind methods,
+		// because we detect the types automatically.
+		arg_names.resize(arg_type_names.size());
```

Comment:
Shouldn't this resize the one that's smaller to the size of the bigger one?

Or at least resize the types one to the size of the name vector, an argument without type can be dealt with (Variant) but a type without any name is weird